### DISCUSSION TITLE
To run Msfconsole occured incompatible library version

### DISCUSSION THREAD

**Discussion by libofuture** [ruby/rubygems#8097] [Q&A]
I have installed latest MetaSploit Framework.  Date: 01/10/2024
when I run msfconsole, occured an error.   
**below is  details:**

```
E:/metasploit-framework/embedded/lib/ruby/3.2.0/monitor.rb:90:in `require': incompatible library version - E:/metasploit-framework/embedded/lib/ruby/3.2.0/x64-mingw-ucrt/monitor.so (LoadError
        from E:/metasploit-framework/embedded/lib/ruby/3.2.0/monitor.rb:90:in `<top (required)>'
        from <internal:E:/metasploit-framework/embedded/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:9:in `require'
        from <internal:E:/metasploit-framework/embedded/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:9:in `<top (required)>'
        from E:/metasploit-framework/embedded/lib/ruby/site_ruby/3.2.0/rubygems.rb:1380:in `eval'
        from E:/metasploit-framework/embedded/lib/ruby/site_ruby/3.2.0/rubygems.rb:1380:in `<top (required)>'
        from <internal:gem_prelude>:2:in `require'
        from <internal:gem_prelude>:2:in `<internal:gem_prelude>'
```

I searched question in BING. some answer is  uninstall ruby but it's not work. who can help me. Thank you very very very much.

**Comment by deivid-rodriguez** [comment]
Hi! Unfortunately, I don't know why reinstalling Ruby did not work, it should've worked.

**Comment by deivid-rodriguez** [comment]
Hi again! I found an issue that could be related to your problem, and I'll be shipping a fix soon. But of course, Metasploit will need to vendor the fixed version before it can get fixed on your side.

**Comment by deivid-rodriguez** [comment]
The patch fixing this was released as RubyGems 3.5.22. If you're still having this issue, I suggest to request Metasploit framework to upgrade their RubyGems version. Thanks!
